stdlib/_ast.pyi

```diff
@@ -27,126 +29,180 @@ if sys.version_info >= (3, 8):
     class type_ignore(AST): ...
 
     class TypeIgnore(type_ignore):
+        if sys.version_info >= (3, 10):
+            __match_args__ = ("lineno", "tag")
         tag: str
 
     class FunctionType(mod):
+        if sys.version_info >= (3, 10):
+            __match_args__ = ("argtypes", "returns")
         argtypes: list[expr]
         returns: expr
 
 class Module(mod):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("body", "type_ignores")
     body: list[stmt]
     if sys.version_info >= (3, 8):
         type_ignores: list[TypeIgnore]
 
 class Interactive(mod):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("body",)
     body: list[stmt]
 
 class Expression(mod):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("body",)
     body: expr
 
 class stmt(AST): ...
 
 class FunctionDef(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("name", "args", "body", "decorator_list", "returns", "type_comment")
     name: _identifier
     args: arguments
     body: list[stmt]
     decorator_list: list[expr]
     returns: expr | None
 
 class AsyncFunctionDef(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("name", "args", "body", "decorator_list", "returns", "type_comment")
     name: _identifier
     args: arguments
     body: list[stmt]
     decorator_list: list[expr]
     returns: expr | None
 
 class ClassDef(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("name", "bases", "keywords", "body", "decorator_list")
     name: _identifier
     bases: list[expr]
     keywords: list[keyword]
     body: list[stmt]
     decorator_list: list[expr]
 
 class Return(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("value",)
     value: expr | None
 
 class Delete(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("targets",)
     targets: list[expr]
 
 class Assign(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("targets", "value", "type_comment")
     targets: list[expr]
     value: expr
 
 class AugAssign(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("target", "op", "value")
     target: expr
     op: operator
     value: expr
 
 class AnnAssign(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("target", "annotation", "value", "simple")
     target: expr
     annotation: expr
     value: expr | None
     simple: int
 
 class For(stmt):
+    if sys.version_info >= (3, 10):
+        __match_args__ = ("target", "iter", "body", "orelse", "type_comment")
```

Comment:
There is no `type_comment` attribute in the stub, seems like we're missing it.

Reply:
It's inherited from `_ast.AST`, of which `_ast.stmt` is a subclass:

https://github.com/python/typeshed/blob/f40747f81df67098543ddf520194a6859e4ddf02/stdlib/_ast.pyi#L12-L22

Reply:
Thanks, I missed that

Reply:
So did I the first time I looked :)